### ISSUE TITLE
How to duplicate a versionable object?

### ISSUE THREAD

**Issue by ezheidtmann** [swisscom/cleanerversion#68]
I have an object X, which is a Versionable model. I want to make a new object Y that has all the same field values as X, but has an unrelated history (i.e. no history). 

In projects without versioning, I can do this:

```python
o = MyModel.objects.get(name='x')
o.id = None
o.pk = None
o.save()
```

But a similar approach gets me an integrity error with CleanerVersion:

`IntegrityError: NOT NULL constraint failed: mymodel.id`

Any advice? I'll post back here when/if I find a solution. Thanks!

**Comment by maennel** [comment]
Hi Evan,

Try out the following:

o.id = o.identity = uuid4()
# let's give the object a new id
o.version_start_date = o.version_birth_date = now()
# and let's set its start date to the current timestamp
o.version_end_date = None
# Ensure the new object to be current
o.save()

As a generic approach I could imagine to have the above code inside a
"duplicate" method on Versionables.

Thanks,
-Manuel

**Comment by ezheidtmann** [comment]
Thanks! This works great. I did it like this:

```python
import six, uuid
import versions.models

class MyVersionable(Versionable):
    def new_identity(self):
        self.id = self.identity = six.u(str(uuid.uuid4()))
        self.version_start_date = self.version_birth_date = versions.models.get_utc_now()
        self.version_end_date = None
        return self

    class Meta:
        abstract = True
```

I am not sure what to call this method: it doesn't make a duplicate; instead, it changes the current object into its own duplicate. And I didn't want to call `save()` because I've ended up in tough situations in the past trying to customize the `save()` method. So I called it `new_identity` because it gives the current object a new identity.

Happy to do a PR if you are OK with this method name, or suggest a different one.

**Comment by brki** [comment]
How about ``sever()`` or, less cuttingly, ``detach()``?

It could accept a boolean parameter ``save``, defaulting to False, that would control whether or not it is saved immediately.

**Comment by ezheidtmann** [comment]
Excellent name! I left out the `save` parameter because you can achieve that with `o.detach().save()`. Probably should have tests for this thing.

**Comment by brki** [comment]
Thanks for the contribution! This has been included in v1.4.5.
